### DISCUSSION TITLE
CardPayment Brick component bug

### DISCUSSION THREAD

**Discussion by joxpulp** [mercadopago/sdk-react#48] [General]
Hi, I was trying to do a test payment and i notice that with this card numbers:

- 031755734530604 (15 length)
- 12554545454545 (14 length)

the card number input does not show an error message in the form:

![Captura de pantalla 2023-05-15 a la(s) 11 49 20 a  m](https://github.com/mercadopago/sdk-react/assets/73070103/fb7d1be9-724b-46fe-87e7-59ce2702295b)

 Then when im trying to do a purchase, in the console this error is shown, this confuse users because they dont know where the error is unless they open devtools and see the console: 
![Captura de pantalla 2023-05-15 a la(s) 11 48 13 a  m](https://github.com/mercadopago/sdk-react/assets/73070103/40df5033-b77f-491a-bf54-a0c70d76e9ed)

**Comment by AleFossati** [comment]
Hello,

Recently we have added a validation for this case, and for me it shows an error on the brick for the card numbers you provided.

<img width="602" alt="image" src="https://github.com/mercadopago/sdk-react/assets/104934463/d8ecf51d-31b2-431f-8ef5-32a528eaa59f">

Can you please try it again and tell us if it still persists?
We know that this generic error message is not the best solution to the user and we are already working on a better feedback.

**Reply by joxpulp** [reply]
I notice that this generics error messages does not show when using hidePaymentButton on true, since we are using a custom payment button for our app, the problem still exist, you can see this behavior here too: 

https://www.mercadopago.com.ar/developers/en/live-demo/card-payment-brick

**Reply by AleFossati** [reply]
Okay, with the `hidePaymentButton = true` I can see the bug.
We'll work on that and update this discussion with the solution.
Thanks for your feedback!

**Reply by joxpulp** [reply]
Thanks :D

**Comment by ghost** [comment]
@joxpulp hello! Sorry for the late reply, we discussed your ticket with the team and I'd like to share an update with you...

The conclusion is that we are not going to fix this, because it is not a bug.
This is an expected behavior, because if the integrator wants to use the Card Payment Brick, hide the button and insert some more inputs below, simulating the continuity of the form asking for other fields, he has that possibility.
If we make the error message appear as you described, for integrators who used the strategy mentioned above, an error would appear right in the middle of the form, which is not visually nice.

Therefore, I'm closing this discussion, thank you for bringing up the topic!
